dan-stineback/test/test.js

```diff
@@ -1,84 +1,149 @@
 'use strict';
 
-const fs = require('fs');
-const express = require('express');
 const chai = require('chai');
 const chaiHTTP = require('chai-http');
-const expect = chai.expect;
+const Cat = require('../schema/cats');
+const Dog = require('../schema/dogs');
+
+const mongoose = require('mongoose');
 chai.use(chaiHTTP);
+
+const expect = chai.expect;
 const request = chai.request;
-const server = require(__dirname + '/../lib/server');
-const app = express();
-const bodyParser = require('body-parser');
-const jsonParser = bodyParser.json();
-const myRouter = express.Router();
+const dbPort = process.env.MONGOLAB_URI;
 
-app.use(jsonParser);
-require('../index');
+process.env.MONGOLAB_URI = 'mongodb://localhost/test_db';
+require('../server');
 
-describe('testing path', () => {
-  it('should return GET message', (done) => {
-    request('localhost:3000')
-    .get('/zoots')
-    .end((err, res) => {
-      expect(err).to.eql(null);
-      expect(res).to.have.status(200);
-      expect(res.text).to.eql('Zoots Zoots. ');
-      done();
+
+describe('Cat test', () => {
+  after((done)=> {
+    process.env.MONGOLAB_URI = dbPort;
+    mongoose.connection.db.dropDatabase(()=>{
+        done();
     });
   });
-  it('should write POST mesage', (done) => {
+  it('should get a list of cats', (done) => {
     request('localhost:3000')
-    .post('/zoots')
-    .send({})
-    .end((err, res) => {
-      expect(err).to.eql(null);
-      expect(res).to.have.status(200);
-      expect(res.text).to.eql(('{"message":"This is some sweet sweet JSON. "}'));
-      done();
-    });
+      .get('/cats/')
+      .end((err, res)=>{
+        expect(err).to.eql(null);
+        expect(Array.isArray(res.body)).to.eql(true);
+        done();
+      });
   });
-  it('should write PUT message', (done) => {
+  it('should post a cats', (done) => {
+    let vic = {name: 'Vic', size: 'large'};
     request('localhost:3000')
-    .put('/zoots')
-    .send({})
-    .end((err, res) => {
-      expect(err).to.eql(null);
-      expect(res).to.have.status(200);
-      expect(res.text).to.eql('{"message":"Party Under Trees JSON "}');
-      done();
+      .post('/cats/')
+      .send(vic)
+      .end((err, res)=>{
+        expect(err).to.eql(null);
+        expect(res).to.have.status(200);
+        expect(res.body.name).to.eql('Vic');
+        expect(res.body).to.have.property('_id');
+        expect(res.body.size).to.eql('large');
```

Comment:
fails because dogs have name, age not name, size properties - I see 2 test failures one for cats and one for dogs